packages/use-dataloader/src/DataLoaderProvider.js

```diff
@@ -0,0 +1,148 @@
+import PropTypes from 'prop-types'
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useRef,
+} from 'react'
+
+export const DataLoaderContext = createContext()
+
+const DataLoaderProvider = ({ children }) => {
+  const cachedData = useRef({})
+  const reloads = useRef({})
+
+  const setCachedData = useCallback(compute => {
+    if (typeof compute === 'function') {
+      cachedData.current = compute(cachedData.current)
+    } else {
+      cachedData.current = compute
+    }
+  }, [])
+
+  const setReloads = useCallback(compute => {
+    if (typeof compute === 'function') {
+      reloads.current = compute(reloads.current)
+    } else {
+      reloads.current = compute
+    }
+  }, [])
+
+  const addCachedData = useCallback(
```

Comment:
addCachedData should be implicite and use everytime we make a load to keep fresh data in the cache.